Update REPORT_A_BUG.yml

jeddy3 opened the pull request:
<!-- Each pull request must be associated with an open issue unless it's a documentation fix. If a corresponding issue does not exist, please create one so we can discuss the change first. -->

<!-- Please answer the following. We close pull requests that don't. -->

> Which issue, if any, is this issue related to?

Follow on from https://github.com/stylelint/stylelint/pull/6274

> Is there anything in the PR that needs further explanation?

Linking to the changelog no longer makes sense. People can see if the bug was reported previously (and fixed) by searching the our issues.

JounQin commented:
I don't quite sure how is this related to changesets, But sure searching issues without changelog should always work already.

ybiquitous commented on .github/ISSUE_TEMPLATE/REPORT_A_BUG.yml:
```diff
@@ -5,11 +5,9 @@ body:
-        Before posting, please check that the bug hasn't already been:
-        * [fixed in the next release](https://github.com/stylelint/stylelint/blob/main/CHANGELOG.md)
-        * [discussed previously](https://github.com/search/advanced?r=stylelint%2Fstylelint&type=Issues)
+        Please [search our issues](https://github.com/search/advanced?r=stylelint%2Fstylelint&type=Issues) to check that the bug hasn't already been reported.
```
[suggest] When the advance search, the `type` parameter seems unnecessary:

```suggestion
        Please [search our issues](https://github.com/search/advanced?r=stylelint%2Fstylelint) to check that the bug hasn't already been reported.
```

Or, if the normal search, using `type` makes sense:

```suggestion
        Please [search our issues](https://github.com/search?q=repo%3Astylelint%2Fstylelint&type=issues) to check that the bug hasn't already been reported.
```

ybiquitous reviewed (approved):
LGTM 👍🏼 

I've left a suggestion about the search URL, but feel free to ignore it. (it's trivial)

ybiquitous reviewed (approved):
Thanks! 👍🏼

jeddy3 commented:
> I don't quite sure how is this related to changesets,

We linked to the changelog in our issue template so that people can easily see if the bug they were going to report was already fixed but not yet released, i.e. by looking at the `## Head` section. We added this link as people used to open duplicate issues.

The changelog [no longer contains this info](https://github.com/stylelint/stylelint/pull/6274/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed) as its now split into separate files in the `.changesets` folder. Searching is now the easiest way to see if a bug has been fixed but not yet released.
